conformance/lang/expressions/literals/string_literal.balt

```diff
@@ -0,0 +1,372 @@
+Test-Case: output
+Description: Test basic string literals.
+Labels: string
+
+public function main() {
+    string a = "Ballerina";
+    io:println(a); // @output Ballerina
+
+    string b = "1234";
+    io:println(b); // @output 1234
+
+    string c = "{name: Kevin}";
+    io:println(c); // @output {name: Kevin}
+
+    io:println("+_)(*&^%$#@!"); // @output +_)(*&^%$#@!
+}
+
+Test-Case: parser-error
+Description: Test invalid usage of line break in string literal.
+Labels: string
+
+public function main() {
+    string  a = "First String
+                Second String"; // @error line break is not allowed in string value
+}
+
+Test-Case: parser-error
+Description: Test invalid usage of `\` in string literal.
+Labels: string
+
+public function main() {
+    string a = "\"; // @error invalid usage of backslash
+}
+
+Test-Case: parser-error
+Description: Test invalid usage of `"` in string literal.
+Labels: string
+
+public function main() {
+    string  a = """; // @error double quote is not allowed in string value
+}
+
+Test-Case: output
+Description: Test string single escape.
+Labels: string, StringSingleEscape
+
+public function main() {
+    io:println("Hello \t world"); // @output Hello    world
+
+    io:println("Hello \nworld!"); // @output Hello
+                                  // @output world!
+
+    io:println("12345 world \rHello"); // @output Hello world
+
+    io:println("2 \\ two"); // @output 2 \ two
+
+    io:println("\"StringValue\""); // @output "StringValue"
+}
+
+Test-Case: output
+Description: Test using `\` to esacape numeric escape.
+Labels: string
+
+public function main() {
+    io:println("\\u{61}"); // @output \u{61}
+}
+
+Test-Case: error
+Description: Test invalid string escape sequence.
+Labels: string
+
+public function main() {
+    string b = "\Ballerina"; // @error `\B` is not a valid escape sequence
```

Comment:
Let's also update the binding patterns wherever applicable to avoid unused variables.
```suggestion
    string _ = "\Ballerina"; // @error `\B` is not a valid escape sequence
```